docs/examples/spectrum/average_photon_energy.py

```diff
@@ -0,0 +1,197 @@
+# from inspection how the relative distribution of light changes as a function
+# of wavelength. We can normalise the spectral irradiance curves to visualise
+# this shift in the shape of the spectrum over the course of the day. In
+# this example, we normalise by dividing each spectral irradiance value by the
+# total broadband irradiance, which we calculate by integrating the entire
+# spectral irradiance distribution with respect to wavelength.
+
+spectral_poa = spectra_components['poa_global']
+wavelength = spectra_components['wavelength']
+
+broadband_irradiance = trapezoid(spectral_poa, wavelength, axis=0)
+
+spectral_poa_normalised = spectral_poa / broadband_irradiance
+
+# Plot the normalised spectra
+plt.figure()
+plt.plot(wavelength, spectral_poa_normalised)
+plt.xlim(200, 2700)
+plt.ylim(0, 0.0018)
+plt.ylabel(r"Normalised Irradiance (nm⁻¹)")
+plt.xlabel(r"Wavelength (nm)")
+time_labels = times.strftime("%H:%M")
+labels = [
+    "{}, AM={:0.02f}".format(*vals)
+    for vals in zip(time_labels, relative_airmass)
+]
+plt.legend(labels)
+plt.show()
+
+# %%
+# We can now see from the normalised irradiance curves that at the start and
+# end of the day, the spectrum is red shifted, meaning there is a greater
+# proportion of longer wavelength radiation. Meanwhile, during the middle of
+# the day, there is a greater prevalence of shorter wavelength radiation — a
+# blue shifted spectrum.
+#
+# How can we quantify this shift? This is where the average photon energy comes
+# into play.
+
+# %%
+# Calculating the average photon energy
+# -------------------------------------
+# To calculate the APE, first we must convert our output spectra from the
+# simulation into a compatible input for
+# :py:func:`pvlib.spectrum.average_photon_energy`. Since we have more than one
+# spectral irradiance distribution, a :py:class:`pandas.DataFrame` is
+# appropriate. We also need to set the column headers as wavelength, so each
+# row is a single spectral irradiance distribution. It is important to remember
+# here that the resulting APE values depend on the integration limits, i.e.
+# the wavelength range of the spectral irradiance input. APE values are only
+# comparable if calculated between the same integration limits. In this case,
+# our APE values are calculated between 300nm and 4000nm.
+
+spectra = pd.DataFrame(spectral_poa).T  # convert to dataframe and transpose
+spectra.index = time_labels  # add time index
+spectra.columns = wavelength  # add wavelength column headers
+
+ape = spectrum.average_photon_energy(spectra)
+
+# %%
+# We can update the normalised spectral irradiance plot to include the APE
+# value of each spectral irradiance distribution in the legend. Note that the
+# units of the APE are electronvolts (eV).
+
+plt.figure()
+plt.plot(wavelength, spectral_poa_normalised)
+plt.xlim(200, 2700)
+plt.ylim(0, 0.0018)
+plt.ylabel(r"Normalised Irradiance (nm⁻¹)")
+plt.xlabel(r"Wavelength (nm)")
+time_labels = times.strftime("%H:%M")
```

Comment:
Maybe consider formatting the timestamp to "09h", "10h", ... to avoid the extra `:00` everywhere.